test/jdk/javax/swing/JButton/4796987/bug4796987.java

```diff
@@ -27,66 +27,89 @@
  * @key headful
  * @requires (os.family == "windows")
  * @summary XP Only: JButton.setBorderPainted() does not work with XP L&F
- * @author Alexander Scherbatiy
  * @library ../../regtesthelpers
- * @library /test/lib
- * @modules java.desktop/com.sun.java.swing.plaf.windows
- *          java.desktop/sun.awt
- * @build jdk.test.lib.OSVersion jdk.test.lib.Platform
  * @build Util
  * @run main bug4796987
  */
 
-import jdk.test.lib.Platform;
-import jdk.test.lib.OSVersion;
-import java.awt.*;
-import javax.swing.*;
-import com.sun.java.swing.plaf.windows.WindowsLookAndFeel;
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
 
 public class bug4796987 {
 
     private static JButton button1;
     private static JButton button2;
     private static JFrame frame;
+    private static JPanel panel;
 
     public static void main(String[] args) throws Exception {
         try {
-            if (Platform.isWindows()
-                && OSVersion.current().equals(OSVersion.WINDOWS_XP)) {
-                UIManager.setLookAndFeel(new WindowsLookAndFeel());
-                testButtonBorder();
-            }
+            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
```

Comment:
maybe we can check all installed L&Fs?

Reply:
Even for classic windows?

Reply:
> Even for classic windows?

I guess the test is applicable to the classic Windows theme, yet [JDK-4796987](https://bugs.openjdk.org/browse/JDK-4796987) states, <q cite="https://bugs.openjdk.org/browse/JDK-4796987#descriptionmodule-label">This problem does not occur if desktop themes descending from the classic (Windows 2000) look are used.</q>

I'm quite reluctant about adding another case to the test.

Let me know if you think it's necessary.